packages/web-components/src/select/select.ts

```diff
@@ -0,0 +1,206 @@
+import { attr, FASTElement, observable } from '@microsoft/fast-element';
+import { ARIAGlobalStatesAndProperties } from '@microsoft/fast-foundation';
+import { SelectSize } from './select.options.js';
+
+/**
+ * The base class used for constructing a fluent-slider custom element
+ * @public
+ */
+export class Select extends FASTElement {
+  constructor() {
+    super();
+    this.handleChange = this.handleChange.bind(this);
+    this.handleInput = this.handleInput.bind(this);
+    Object.assign(this, new ARIAGlobalStatesAndProperties()); // Assigns ARIA global states and properties to the element
+  }
+
+  // Attributes
+  /**
+   * The name of the select element
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: name
+   */
+  @attr
+  public name?: string;
+
+  /**
+   * The required boolean attribute of the select element
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: required
+   */
+  @attr({ mode: 'boolean' })
+  public required?: boolean;
+
+  /**
+   * The optional label of the select
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: label
+   */
+  @attr
+  public label?: string;
+
+  /**
+   * The autofocus attribute of the select
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: autofocus
+   */
+  @attr({ attribute: 'autofocus', mode: 'boolean' })
+  public autofocusAttr?: boolean;
```

Comment:
Is there a reason that the attribute and the property need to be different here?

Reply:
I was a bit confused here. I renamed the property because it was yelling at me about it being undefined in FASTElement: `Property 'autofocus' in type 'Select' is not assignable to the same property in base type 'FASTElement'. Type 'boolean | undefined' is not assignable to type 'boolean'. Type 'undefined' is not assignable to type 'boolean'.`

If I remove the `?` like the [text-area component does in Fast Foundation](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-foundation/src/text-area/text-area.ts#L57-L58) it gives me this error: `Property 'autofocus' has no initializer and is not definitely assigned in the constructor.ts(2564)` I'm at a loss on how to fix this.

Reply:
Something else is likely going on here, I can pull down and take a look when I have a moment.